crates/turborepo-lib/src/boundaries.rs

```diff
@@ -0,0 +1,443 @@
+            ColorConfig::Auto,
+            true,
+            false,
+            Some(self.source_map.clone()),
+        );
+
+        for diagnostic in self.diagnostics.clone() {
+            match diagnostic {
+                BoundariesDiagnostic::ParseError(_, e) => {
+                    e.clone().into_diagnostic(&handler).emit();
+                }
+                e => {
+                    eprintln!("{:?}", Report::new(e.to_owned()));
+                }
+            }
+        }
+    }
+}
+
+impl Run {
+    pub async fn check_boundaries(&self) -> Result<BoundariesResult, Error> {
+        let packages = self.pkg_dep_graph().packages();
+        let repo = Repository::discover(&self.repo_root()).ok().map(Mutex::new);
+        let mut diagnostics = vec![];
+        let source_map = SourceMap::default();
+        for (package_name, package_info) in packages {
+            if !self.filtered_pkgs().contains(package_name)
+                || matches!(package_name, PackageName::Root)
+            {
+                continue;
+            }
+
+            let package_root = self.repo_root().resolve(package_info.package_path());
+
+            let internal_dependencies = self
+                .pkg_dep_graph()
+                .immediate_dependencies(&PackageNode::Workspace(package_name.to_owned()))
+                .unwrap_or_default();
+            let unresolved_external_dependencies =
+                package_info.unresolved_external_dependencies.as_ref();
+
+            let package_diagnostics = self
+                .check_package(
+                    &repo,
+                    &package_root,
+                    &package_info.package_json,
+                    internal_dependencies,
+                    unresolved_external_dependencies,
+                    &source_map,
+                )
+                .await?;
+
+            diagnostics.extend(package_diagnostics);
+        }
+
+        Ok(BoundariesResult {
+            source_map: Arc::new(source_map),
+            diagnostics,
+        })
+    }
+
+    fn is_potential_package_name(import: &str) -> bool {
+        PACKAGE_NAME_REGEX.is_match(import)
+    }
+
+    /// Either returns a list of errors or a single, fatal error
+    async fn check_package(
+        &self,
+        repo: &Option<Mutex<Repository>>,
+        package_root: &AbsoluteSystemPath,
+        package_json: &PackageJson,
+        internal_dependencies: HashSet<&PackageNode>,
+        unresolved_external_dependencies: Option<&BTreeMap<String, String>>,
+        source_map: &SourceMap,
+    ) -> Result<Vec<BoundariesDiagnostic>, Error> {
+        let files = globwalk::globwalk(
+            package_root,
+            &[
+                "**/*.js".parse().unwrap(),
+                "**/*.jsx".parse().unwrap(),
+                "**/*.ts".parse().unwrap(),
+                "**/*.tsx".parse().unwrap(),
+                "**/*.vue".parse().unwrap(),
+                "**/*.svelte".parse().unwrap(),
```

Comment:
would love a canonical list of all possible `js` extensions! I might have missed one